evalml/pipelines/pipeline_base.py

```diff
@@ -432,38 +434,51 @@ def graph_json(self):
         y_edges specifies from which component target data is being passed.
         This can be used to build graphs across a variety of visualization tools.
         Template:
-        {"Nodes": {"component_name": {"Name": class_name, "Attributes": parameters_attributes}, ...}},
+        {"Nodes": {"component_name": {"Name": class_name, "Parameters": parameters_attributes}, ...}},
         "x_edges": [[from_component_name, to_component_name], [from_component_name, to_component_name], ...],
         "y_edges": [[from_component_name, to_component_name], [from_component_name, to_component_name], ...]}
 
         Returns:
             dag_json (str): A serialized JSON representation of a DAG structure.
         """
-        nodes = {
-            comp_: {"Attributes": att_.parameters, "Name": att_.name}
-            for comp_, att_ in self.component_graph.component_instances.items()
-        }
-
-        x_edges = self.component_graph._get_edges(
+        nodes = {}
+        for comp_, att_ in self.component_graph.component_instances.items():
+            param_dict = {}
+            for param, val in att_.parameters.items():
+                # Can't JSON serialize components directly, have to split them into name and parameters
+                if isinstance(val, ComponentBase):
+                    param_dict[f"{param}_name"] = val.name
+                    param_dict[f"{param}_parameters"] = val.parameters
+                else:
+                    if isinstance(val, np.int64):
```

Comment:
@eccabay I think we'll also have a problem with np.bool or any non-json-serializable parameter. Users can do pretty pathological things, like the following:

```python
from evalml.pipelines import RegressionPipeline
import pandas as pd
import numpy as np
import json
import pytest

X = pd.DataFrame({"a": [1, 2, 2, 1],
                  "b": ["2021-01-01", "2021-01-02", "2021-01-03", "2021-01-04"]})
y = pd.Series([1, 2, 3])

pl = RegressionPipeline(["Delayed Feature Transformer"],
                        parameters={"Delayed Feature Transformer": {"delay_features": np.bool_(False)}})
pl.fit(X, y)
with pytest.raises(TypeError, match="Object of type bool_ is not JSON serializable"):
    pl.graph_json()
```

or 

```python
from evalml.pipelines import RegressionPipeline
import pandas as pd
import json
import pytest

X = pd.DataFrame({"a": [1, 2, 3, 3, 4, 4, None],
                  "b": range(7)})
y = pd.Series([1, 2, 3])

pl = RegressionPipeline(["Imputer"],
                        parameters={"Imputer": {"numeric_fill_value": np.float32(2),
                                                "numeric_impute_strategy": "constant"}})
pl.fit(X, y)
with pytest.raises(TypeError, match="Object of type float32 is not JSON serializable"):
    pl.graph_json()
```

Users could also define their own components that take in pd.Series or np arrays which are also not json serializable. 

Throwing out an alternative, we could just return the python dict and let the user serialize it if they want to serialize it?

Reply:
You raise a very interesting question. 

You're completely correct in that users can set parameters that will break this code. I can pretty easily update this code so it catches any `numpy.number` dtypes, but there's no good way to safely convert and serialize all non-serializable cases. However, with these changes we finally support serializing output from any *default* AutoML search, and that should be guaranteed. 

We have a specific use case driving the need for this function, I'd hate to take away its functionality purely because it may not be able to JSON serialize non-JSON serializable types. I would rather add a bit of hardening here to throw our own error if a pipeline is not serializable, rather than change the underlying goal of this method.